gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/analytics_engine/permissions/ApiAnalyticsQueryFilterDecoratorImpl.java

```diff
@@ -44,44 +56,136 @@
+    private static final String ORGANIZATION_ADMIN = RoleScope.ORGANIZATION.name() + ':' + SystemRole.ADMIN.name();
 
-    private final ApiAuthorizationService apiAuthorizationService;
+    private static final String UNKNOWN_APPLICATION = "Unknown";
 
-    private final PermissionService permissionService;
+    private final ApiSearchService apiSearchService;
+
+    private final ApplicationService applicationSearchService;
 
     @Override
-    public Map<String, API> getAllowedApis() {
-        var allowedApis = new HashMap<String, API>();
-
-        var executionContext = GraviteeContext.getExecutionContext();
-        if (isEnvironmentAdmin()) {
-            var apiIds = apiAuthorizationService.findIdsByEnvironment(executionContext.getEnvironmentId());
-            apiIds.forEach(id -> allowedApis.put(id, new API("")));
-            return allowedApis;
+    public MetricsContext getFilteredContext(MetricsContext context, List<Filter> requestFilters) {
+        var allowedApis = getAllowedApis(requestFilters);
+        var allowedApplications = getAllowedApplications(requestFilters);
+        var updatedFilters = updateContextFilters(context.filters(), allowedApis);
+
+        return context.withApiNamesById(allowedApis).withApplicationNameById(allowedApplications).withFilters(updatedFilters);
+    }
+
+    private Map<String, String> getAllowedApis(List<Filter> filters) {
+        QueryBuilder<ApiEntity> queryBuilder = QueryBuilder.create(ApiEntity.class);
+
+        var apiIds = getIdsFromFilters(filters, API);
+        if (!(apiIds.isEmpty())) {
+            queryBuilder.setFilters(Map.of("id", apiIds));
+        }
+
+        boolean admin = isAdmin();
+        Principal userPrincipal = getUserPrincipal();
+        String name = userPrincipal.getName();
+        ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        Page<GenericApiEntity> search = apiSearchService.search(
+            executionContext,
+            name,
+            admin,
+            queryBuilder,
+            new PageableImpl(0, Integer.MAX_VALUE),
+            false,
+            true
+        );
+
+        List<GenericApiEntity> content = search.getContent();
+
+        return mapApiIdsToNames(content);
+    }
+
+    private Map<String, String> getAllowedApplications(List<Filter> filters) {
+        var query = new ApplicationQuery();
+
+        var applicationIds = getIdsFromFilters(filters, APPLICATION);
+        if (!(applicationIds.isEmpty())) {
+            query.setIds(applicationIds);
+        }
+
+        if (!isAdmin()) {
+            query.setUser(getAuthenticatedUser());
+        }
+
+        List<ApplicationListItem> content = applicationSearchService
+            .search(GraviteeContext.getExecutionContext(), query, null, null)
+            .getContent();
+        return mapApplicationIdsToNames(content);
+    }
```

Comment:
This is not needed in this user context. We only need to retrieve APIs the user can manage to enforce permissions.